torrent/torrent_file.py

```diff
@@ -1,64 +1,83 @@
 
 #classes
 
-class TorrentFile(object):
+class TorrentFile():
     def __init__(self, struct_dict):
+        self.multi_file = struct_dict['multi_file']
         self.path = struct_dict['path']
+        self.od_data = struct_dict['orderd_content'] # all the data in OrderdDict object
         self.trackers = struct_dict['trackers']
         self.info_hash = struct_dict['info_hash']
-        self.files = struct_dict['files']
+        if self.multi_file:
+            self.files = struct_dict['files']
         self.length = struct_dict['length']
         self.piece_length = struct_dict['piece_length']
         self.name = struct_dict['name']
+        if type(self.name) == bytes:
+            self.name = self.name.decode('utf-8')
         self.pieces = struct_dict['pieces']
+        
         self.raw_content = struct_dict['raw']
+        self.peer_id = '-PC0001-' + ''.join([str(randint(0, 9)) for _ in range(12)])
+
+    def __repr__(self):
+        return f"torrent_file({self.name})"
 
-class Tracker(object):
+
+class Tracker():
     #TODO think if this is the right place hold this object and if the the future the tracker going to
     #manage the connection
     """
     the object contain 2 types of trackers udp or http trackers
     url(domain):str, path:str, type:'http' or 'udp, port:'int'
     """
+    def __init__(self, tracker_type, url, path,  port=80):
+        self.schema = tracker_type
+        self.url = url
+        self.path = path
+        self.port = port
+
+        if tracker_type == 'https':
+            self.tracker_type = 'http'
+        else:
+            self.tracker_type = tracker_type 
+
+    def http_url(self):
+        return f'{self.schema}://{self.url}:{self.port}{self.path}'
+
+    def __repr__(self):
+        return f'tracker({self.schema}:{self.url}:{self.port}{self.path})'
+
+    def __hash__(self):
+        return hash(self.url)
+
+    def __eq__(self, other):
+        return (self.__class__ == other.__class__ and self.url == other.url and self.tracker_type == other.tracker_type and self.port == other.port)
+
+    def __ne__(self, other):
+        return not self.__eq__(self, other)
     
-    def __init__(self, url,port, path, tracker_type):
-       self.url = url
-       self.path = path
-       self.tracker_type = tracker_type
 
 
 
 #help functions
-def od_get_key(od, key, mandat=False):
-    """
-    the function get OrederdDict object key and mandet
-    the function try to get back the data from the od 
-    if it mandat raise the error if the key is not mandat 
-    return None
-    """
-    if type(od) != OrderedDict:
-        raise TypeError(f'od_get_key get {type(od)} and not OrderdDict object')
-    try:
-        return od[key]
-    except KeyError as e:
-        if mandat:
-            raise e
-        else:
-            return None
-        
 
 #torrent parse functions
 def valid_torrent_path(path):
     # the function valid torrent object path
+    if type(path) != str:
```

Comment:
again, i think you should make the change to python3.8 and add param type validation